### ISSUE TITLE
InfluxDB: Run queries concurently on the backend

### ISSUE THREAD

**Issue by itsmylife** [grafana/grafana#80955]
We run influxql queries in a for loop on the backend and that is not quite performant for requests that contains many queries. That means each query must wait other query to completed. Especially for heavy queries, response time is a problem. We should be able run queries concurrently so we will be able to return responses much faster. 

https://github.com/grafana/grafana/blob/main/pkg/tsdb/influxdb/influxql/influxql.go#L34-L64

**Comment by 1tft** [comment]
We tried to use this feature inside alert rules and activated this feature:

```
[unified_alerting]
max_state_save_concurrency = 6

[feature_toggles]
influxdbRunQueriesInParallel = true
```

No success. Alert rule evaluation time is still very high.

We also noticed, that if you use 3 queries inside one alert rule against **different** InfluxDB (InfluxQL) datasources, there is no concurrent run. Evluation time is still sum of query01 response time + query02 response time + query03 response time.
Thats very frustrating.
